src/WebJobs.Extensions.DurableTask/HttpApiHandler.cs

```diff
@@ -212,10 +214,14 @@ public async Task<HttpResponseMessage> HandleRequestAsync(HttpRequestMessage req
                         string entityName = (string)routeValues[EntityNameRouteParameter];
                         string entityKey = (string)routeValues[EntityKeyRouteParameter];
                         EntityId entityId = new EntityId(entityName, entityKey);
-                        if (request.Method == HttpMethod.Get)
+                        if (!string.IsNullOrEmpty(entityKey) && request.Method == HttpMethod.Get)
                         {
                             return await this.HandleGetEntityRequestAsync(request, entityId);
                         }
+                        else if (request.Method == HttpMethod.Get)
+                        {
+                            return await this.HandleListEntitiesRequestAsync(request, entityName);
+                        }

```

Comment:
I suggest we structure this a bit differently:
```csharp
else if (request.Method == HttpMethod.Get)
{
    // TODO: Need a convention for handling the case for a single entity with an empty entity key
    if (!string.IsNullOrEmpty(entityKey))
    {
        return await this.HandleGetEntityRequestAsync(request, entityId);
    }
    else
    {
        return await this.HandleListEntitiesRequestAsync(request, entityName);
    }
}
```
This is a slightly safer way to implement the exact same logic. It's a little easier to understand and it is less likely to be accidentally broken by future if/else condition changes.